projects/packages/connection/src/class-error-handler.php

```diff
@@ -156,29 +156,40 @@ private function __construct() {
 	 * @return void
 	 */
 	public function handle_verified_errors() {
+		// Get all errors (with filter) to see what external plugins have injected
 		$verified_errors = $this->get_verified_errors();
+
+		// Default error codes that should trigger admin notices and React dashboard integration
+		$default_handled_codes = array(
+			'malformed_token',
+			'token_malformed',
+			'no_possible_tokens',
+			'no_valid_user_token',
+			'no_valid_blog_token',
+			'unknown_token',
+			'could_not_sign',
+			'invalid_token',
+			'token_mismatch',
+			'invalid_signature',
+			'signature_mismatch',
+			'no_user_tokens',
+			'no_token_for_user',
+			'invalid_connection_owner',
+		);
+
+		// Automatically include any error codes that aren't in the default list
```

Comment:
As we discussed, I think it's safer to hardcode the error code coming from `wpcomsh` here and also add it in the `$known_errors` list.
The concept of this class is that we only allow verified connection errors to show up.
An exception is the `invalid_connection_owner` which is not verified but is set by the same package so it's safe.
The filter we introduced is enough to allow `wpcomsh` to handle it's own connection errors without storing them but I don't see any other plugin handling connection errors.